synapse/storage/databases/main/devices.py

```diff
@@ -1614,20 +1618,21 @@ async def check_too_many_devices_for_user(self, user_id: str) -> List[str]:
 
         # Now fetch the devices to delete.
         sql = """
-            SELECT DISTINCT device_id FROM devices
```

Comment:
FTR, `(device_id, user_id)` is unique by constraint hence the DISTINCT should indeed be useless.